### ISSUE TITLE
Crash on OS X

### ISSUE THREAD

**Issue by poke1024** [bjorn/tiled#1659]
Tiled crashes, not 100% reproducibly, when its file changed event is triggered on OS X (see below).

For me this seems to happen quite often when I launch a löve script called lurker (https://github.com/rxi/lurker) that monitors the folder the currently open Tiled map resides in (while that map is open in Tiled). That script does not change any map files, but it seems to trigger Tiled::FileSystemWatcher::fileChanged, and from there it seems as if Tiled always crashes. Is this a known bug?

```
Process:               Tiled [28988]
Path:                  /Users/USER/Desktop/*/Tiled.app/Contents/MacOS/Tiled
Identifier:            org.mapeditor.Tiled
Version:               1.0.2 (1.0.2)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Tiled [28988]
User ID:               503

Date/Time:             2017-07-22 16:01:28.152 +0200
OS Version:            Mac OS X 10.12.5 (16F73)
Report Version:        12

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   org.mapeditor.Tiled           	0x0000000109335f84 Tiled::Internal::DocumentManager::checkTilesetColumns(Tiled::Internal::MapDocument*) + 68
1   org.mapeditor.Tiled           	0x00000001093359cf Tiled::Internal::DocumentManager::reloadDocumentAt(int) + 271
2   org.mapeditor.Tiled           	0x00000001093317b2 Tiled::Internal::DocumentManager::fileChanged(QString const&) + 258
3   org.qt-project.QtCore         	0x000000010a50759b QMetaObject::activate(QObject*, int, int, void**) + 731
4   libtiled.1.0.0.dylib          	0x000000010969d0f5 Tiled::FileSystemWatcher::fileChanged(QString const&) + 53
5   libtiled.1.0.0.dylib          	0x000000010965291a Tiled::FileSystemWatcher::onFileChanged(QString const&) + 266
6   org.qt-project.QtCore         	0x000000010a507651 QMetaObject::activate(QObject*, int, int, void**) + 913
7   org.qt-project.QtCore         	0x000000010a45f4f1 0x10a2f0000 + 1504497
8   org.qt-project.QtCore         	0x000000010a507651 QMetaObject::activate(QObject*, int, int, void**) + 913
9   org.qt-project.QtCore         	0x000000010a598950 0x10a2f0000 + 2787664
10  org.qt-project.QtCore         	0x000000010a5006fc QObject::event(QEvent*) + 156
11  org.qt-project.QtWidgets      	0x000000010975da4d QApplicationPrivate::notify_helper(QObject*, QEvent*) + 269
12  org.qt-project.QtWidgets      	0x0000000109760392 QApplication::notify(QObject*, QEvent*) + 5906
13  org.qt-project.QtCore         	0x000000010a4d65c4 QCoreApplication::notifyInternal2(QObject*, QEvent*) + 164
14  org.qt-project.QtCore         	0x000000010a4d71f8 QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) + 600
15  libqcocoa.dylib               	0x000000010c3e0f4e 0x10c3bb000 + 155470
16  libqcocoa.dylib               	0x000000010c3e1811 0x10c3bb000 + 157713
17  com.apple.CoreFoundation      	0x00007fffbd298321 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
18  com.apple.CoreFoundation      	0x00007fffbd27921d __CFRunLoopDoSources0 + 557
19  com.apple.CoreFoundation      	0x00007fffbd278716 __CFRunLoopRun + 934
20  com.apple.CoreFoundation      	0x00007fffbd278114 CFRunLoopRunSpecific + 420
21  com.apple.HIToolbox           	0x00007fffbc7d9ebc RunCurrentEventLoopInMode + 240
22  com.apple.HIToolbox           	0x00007fffbc7d9cf1 ReceiveNextEventCommon + 432
23  com.apple.HIToolbox           	0x00007fffbc7d9b26 _BlockUntilNextEventMatchingListInModeWithFilter + 71
24  com.apple.AppKit              	0x00007fffbad72a54 _DPSNextEvent + 1120
25  com.apple.AppKit              	0x00007fffbb4ee7ee -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 2796
26  com.apple.AppKit              	0x00007fffbad673db -[NSApplication run] + 926
27  libqcocoa.dylib               	0x000000010c3e06bf 0x10c3bb000 + 153279
28  org.qt-project.QtCore         	0x000000010a4d29d1 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) + 401
29  org.qt-project.QtCore         	0x000000010a4d6c35 QCoreApplication::exec() + 341
30  org.mapeditor.Tiled           	0x000000010934fc0a main + 3818
31  libdyld.dylib                 	0x00007fffd29f2235 start + 1
```

**Comment by bjorn** [comment]
Not a known issue, but I'll be sure to investigate as soon as possible. Thanks for the report!

**Comment by manabreak** [comment]
I ran into this same problem when I was working on a map that was in a git repo and stashed the changes. Not sure if this info has any value, but thought to mention it anyway.

**Comment by lzubiaur** [comment]
Hi, I have a similar issue on mac with git. When I edit a map in a git repo and  I switch branch, Tiled sometimes failed to reload the map and crashes.

**Comment by bjorn** [comment]
Fix will be in Tiled 1.0.3, expected tomorrow.
